### DISCUSSION TITLE
Ubuntu 20.04 - pycharm - Firefox not headless

### DISCUSSION THREAD

**Discussion by motovmp1** [seleniumbase/SeleniumBase#964] [Q&A] [answered]
Hello friends

Ubuntu 20.04 and pycharm

Chrome runs headless by default  and --headless works also

But, Firefox always run in GUI mode even if I use --headless in CLI.

I want to run i headless mode

Thanks

VP

**Comment by mdmintz** [comment, marked as answer]
Hello @motovmp1 , based on the open ticket https://bugzilla.mozilla.org/show_bug.cgi?id=1688818 , there might be a Firefox bug with headless mode on Ubuntu 20.04 . The good news is that it works fine on Ubuntu 18.04 - See the Dockerfile: https://github.com/seleniumbase/SeleniumBase/blob/master/Dockerfile
If you find a workaround for headless Firefox on Ubuntu 20.04, please let me know.

**Reply by motovmp1** [reply]
copy that!

**Reply by motovmp1** [reply]
sudo apt-get install firefox xvfb
Install this one and works perfect!

pytest --headed -k test_login_page_dashboard_normal_user_analysis_visible  --html=report.html --dashboard -s  --demo --firefox --headless

**Reply by mdmintz** [reply]
Awesome! So ``xvfb`` does the trick. That's why it's a requirement on the last line of the ReadMe in https://github.com/mdmintz/sbVirtualDisplay , which is used by SeleniumBase as a dependency to activate a virtual display in situations where the default headless mode of browsers doesn't work.

**Reply by motovmp1** [reply]
Thanks a lot...cheers!
